crates/api_crud/src/site/update.rs

```diff
@@ -76,6 +76,24 @@ impl PerformCrud for EditSite {
       }
     }
 
+    if data.private_instance == Some(true)
+      && data
+        .federation_enabled
+        .unwrap_or(local_site.federation_enabled)
+    {
+      return Err(LemmyError::from_message(
+        "cant_enable_private_instance_if_federation_enabled",
+      ));
+    }
+
+    if data.federation_enabled == Some(true)
+      && data.private_instance.unwrap_or(local_site.private_instance)
+    {
+      return Err(LemmyError::from_message(
+        "cant_enable_federation_if_private_instance",
```

Comment:
I dont think these two separate error messages add anything. I would only make a message "cant enable federation and private instance together".